### DISCUSSION TITLE
Notification when packages updates are available

### DISCUSSION THREAD

**Discussion by pedrom34** [MichaIng/DietPi#7289] [Q&A] [answered]
Hey!
I like the fact that dietpi tells me that apt updates are available when I connect through ssh.
But I was wondering... Is there a way to notify my phone with ntfy? I know I can do it via email with apticron or webmin, but I rather do it with ntfy.
So, when updates are available is there a way to add a curl command with the number of packages upgradable to the process so I can have a push alert?

I don't really how to start...

Thanks!

**Comment by MichaIng** [comment, marked as answer]
Forwarding to this feature request: #2663

Generally, you could add a daily cron job with a script like `/etc/cron.daily/zz-notification`, check whether `/run/dietpi/.apt_updates` exists, which contains the number of upgradable packages in case. The check is done, and this file generated, in `/etc/cron.daily/dietpi`, hence any script named alphanumerically after that, will work. Any MTA can then be used to send an email.

**Reply by pedrom34** [reply]
Thanks a lot Micha, That will help me to write a bash script.

I'll post it when I find the time to write it 👍

**Comment by pedrom34** [comment]
I wrote this. It works with ntfy (a self-hosted instance, but I changed the curl command to make it work with https://ntfy.sh), I need to wait for a package update to test if it works IRL.

```
#!/bin/bash

FILE="/run/dietpi/.apt_updates"

if [ -f "$FILE" ]; then
  # Read the number of packages to update from the file
  NB_PACKETS=$(cat "$FILE")

  # Run the apt upgrade -s command and store the output in a variable
  OUTPUT=$(apt upgrade -s 2>/dev/null)

  # Check if the command succeeded
  if [ $? -eq 0 ]; then
    # Extract the list of upgradable packages from the output
    PACKETS=$(echo "$OUTPUT" | grep "^Inst" | awk '{print $2}' | tr '\n' ',' | sed 's/,$//')

    # Send the notification with the number of packages and the list of packages
    curl \
      -H "Title: Package updates available" \
      -H "Tags: eyes" \
      -d "There are $NB_PACKETS packages to update on the server: [$PACKETS]." \
      ntfy.sh/topic
  else
    # The command failed
    echo "The apt upgrade -s command failed for some reason..."
  fi
fi
```

**Reply by Joulinar** [reply]
just played with it, `ntfy` is quite a nice app :)

>```
>  OUTPUT=$(apt upgrade -s 2>/dev/null)
>```

On this, I would recommend using `apt-get` instead of `apt`.

**Reply by pedrom34** [reply]
Yes, ntfy is really cool. Very simple to use and implement, light on resources as well 👍 
I'm using for a lot of things: uptime kuma, acme.sh, backup notifications, update on containers, movie grabbed by radarr, even to share files between my computer and my phone. I used to go with telegram, but self hosted is the way 🫡 

Thanks for the feedback! I'll edit my script!
